clang/lib/Driver/Driver.cpp

```diff
@@ -4995,9 +4995,8 @@ class OffloadingActionBuilder final {
             ActionList CAList;
             CAList.push_back(A);
             ActionList DeviceLinkActions;
-            appendSYCLDeviceLink(CAList, TargetInfo.TC, DA, DeviceLinkActions,
-                                 TargetInfo.BoundArch,
-                                 /*AddOffloadAction=*/true);
+            appendSYCLDeviceLink(CAList, TargetInfo.TC, DeviceLinkActions,
```

Comment:
this comment is not really related to this PR so feel free to respond with lower priority, but i've started working on `-fno-sycl-rdc` with the new offload driver, and it seems like in that case we don't use the `DeviceActionBuilders`, so it makes it somewhat hard to call the SYCL device linking backend. my idea is to call it at `Driver::BuildOffloadingActions`, but to do that it seems we would need to make `appendSYCLDeviceLink` callable without the builders. is that the right idea or am i misunderstanding the flow?

Reply:
That sounds about right.  Typical expectation is for the `appendSYCLDeviceLink` to only occur within the SYCL specific toolchain.  If we require access to this behavior otherwise, we need to pull it out and generalize it, which I would assume may require additional parameters to satisfy any AOT specific behaviors within.